libcst/metadata/tests/test_type_inference_provider.py

```diff
@@ -0,0 +1,259 @@
+                        "location": {
+                            "start": {"line": 11, "column": 32},
+                            "stop": {"line": 11, "column": 40},
+                        },
+                        "annotation": "range",
+                    },
+                    {
+                        "location": {
+                            "start": {"line": 15, "column": 8},
+                            "stop": {"line": 15, "column": 27},
+                        },
+                        "annotation": "typing.Callable(libcst.metadata.example_type_infer.ItemCollector.get_items)[[Named(n, int)], typing.Sequence[libcst.metadata.example_type_infer.Item]]",
+                    },
+                    {
+                        "location": {
+                            "start": {"line": 10, "column": 24},
+                            "stop": {"line": 10, "column": 25},
+                        },
+                        "annotation": "int",
+                    },
+                    {
+                        "location": {
+                            "start": {"line": 6, "column": 22},
+                            "stop": {"line": 6, "column": 23},
+                        },
+                        "annotation": "int",
+                    },
+                    {
+                        "location": {
+                            "start": {"line": 16, "column": 12},
+                            "stop": {"line": 16, "column": 17},
+                        },
+                        "annotation": "typing.Sequence[libcst.metadata.example_type_infer.Item]",
+                    },
+                    {
+                        "location": {
+                            "start": {"line": 17, "column": 4},
+                            "stop": {"line": 17, "column": 8},
+                        },
+                        "annotation": "libcst.metadata.example_type_infer.Item",
+                    },
+                    {
+                        "location": {
+                            "start": {"line": 14, "column": 0},
+                            "stop": {"line": 14, "column": 9},
+                        },
+                        "annotation": "libcst.metadata.example_type_infer.ItemCollector",
+                    },
+                ]
+            },
+        )
+        types = wrapper.resolve(TypeInferenceProvider)
+        m = wrapper.module
+        self_number_attr = cst.ensure_type(
+            cst.ensure_type(
+                cst.ensure_type(
+                    cst.ensure_type(
+                        cst.ensure_type(m.body[1].body, cst.IndentedBlock).body[0],
+                        cst.FunctionDef,
+                    ).body.body[0],
+                    cst.SimpleStatementLine,
+                ).body[0],
+                cst.Assign,
+            )
+            .targets[0]
+            .target,
+            cst.Attribute,
+        )
+        self.assertEqual(types[self_number_attr], "int")
```

Comment:
This is a rather large example, but then you only inspect two types on it.

I'd either test more things, or strip down the size of the example.

Reply:
Some more asserts are added after your review.